### ISSUE TITLE
Unable to use with sonarr or nzbhydra

### ISSUE THREAD

**Issue by Qballjos** [jgeusebroek/docker-spotweb#16]
Hi I'm using this docker container on my unraid server but when i want to use it with sonarr or nzbhydra2 it gives an error like this.

Error message from sonarr:

`http://192.168.2.80:580/api?t=tvsearch&cat=5030,5040&extended=1&apikey=(removed)&offset=0&limit=100&tvmazeid=36471&season=1&ep=1: On TVSearch no q, tvmaze or rid parameter present`

I have found a thread on the spot web github: 
https://github.com/spotweb/spotweb/issues/367

is there a way to fix this?

**Comment by jgeusebroek** [comment]
I added a possible fix, according to https://github.com/spotweb/spotweb/wiki/Spotweb-als-Newznab-Provider.

Can you let me know if this fixes the issue for you? A new images is being build as I type this.

**Comment by Qballjos** [comment]
Thnx for the response!, I have tested this but it didn't solve the issue...

`NzbDrone.Core.Indexers.Newznab.NewznabException: On TVSearch no q, tvmaze or rid parameter present
  at NzbDrone.Core.Indexers.Newznab.NewznabRssParser.PreProcess (NzbDrone.Core.Indexers.IndexerResponse indexerResponse) [0x000d4] in M:\BuildAgent\work\5d7581516c0ee5b3\src\NzbDrone.Core\Indexers\Newznab\NewznabRssParser.cs:47 
  at NzbDrone.Core.Indexers.RssParser.ParseResponse (NzbDrone.Core.Indexers.IndexerResponse indexerResponse) [0x0000d] in M:\BuildAgent\work\5d7581516c0ee5b3\src\NzbDrone.Core\Indexers\RssParser.cs:55 
  at NzbDrone.Core.Indexers.HttpIndexerBase`1[TSettings].FetchPage (NzbDrone.Core.Indexers.IndexerRequest request, NzbDrone.Core.Indexers.IParseIndexerResponse parser) [0x00050] in M:\BuildAgent\work\5d7581516c0ee5b3\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:308 
  at NzbDrone.Core.Indexers.HttpIndexerBase`1[TSettings].FetchReleases (System.Func`2[T,TResult] pageableRequestChainSelector, System.Boolean isRecent) [0x000a0] in M:\BuildAgent\work\5d7581516c0ee5b3\src\NzbDrone.Core\Indexers\HttpIndexerBase.cs:143 `

**Comment by o0donder0o** [comment]
Thanks for providing this docker container!

I've noticed the same issue. When i do a manual search nothing shows up at first from the Spotweb indexer. 

Oddly enough when i press "test" within the Spotweb indexer settings and do a manual (interactive) search again the first api key seems to work and results are shown. Pressing download doesn't seem to work at the other hand, error message "download from indexer failed" is shown. Consecutive api search queries also do show any results from Spotweb, perhaps the result from Spotweb is excluded by Sonarr. 

I've tried to manually edit the .htaccess file and did notice a difference then the Spotweb instruction refers to. So manually changing .htaccess and restarting apache2 didn't seem to fix it.

Any chance your still going to look into this? I've setup Spotweb mainly as indexer for radarr and sonarr, so i hope i will be fixed. If i can help with anything let me know.

**Comment by jgeusebroek** [comment]
I have tested it, and the fault seems not to be on the Spotweb side. It does return a query result.

Try having a look at the apache2 logfile in spotweb.
`docker exec -ti spotweb_instance tail -f /var/log/apache2/other_vhosts_access.log`

**Comment by o0donder0o** [comment]
After digging more into this i think your right. I've placed spotweb behind a reverse proxy. Had to set the following option to make this work. Layout was messed up without it.
https://github.com/spotweb/spotweb/wiki/Installation-tips-and-troubleshooting#using-a-seperate-ssl-proxy

This probably interferes when pointing sonarr to spotweb directly on ip-address (both containers within same docker network).

For reference here are my logs.

**Logs from Sonarr**

19-12-4 23:20:30.5|Error|ReleaseModule|Downloading nzb failed

[v3.0.3.659] NzbDrone.Core.Exceptions.ReleaseDownloadException: Downloading nzb failed ---> System.Net.WebException: Error: ConnectFailure (Connection refused): 'https://123.1.2.3/api?t=g&id=123456678asdasdasdasdas@spot.net&apikey=(removed) ---> System.Net.WebException: Error: ConnectFailure (Connection refused) ---> System.Net.Sockets.SocketException: Connection refused
  at System.Net.Sockets.SocketAsyncResult.CheckIfThrowDelayedException () [0x00014] in <91935ad653254a93b9d73a9f8f2f7a2d>:0 
  at System.Net.Sockets.Socket.EndConnect (System.IAsyncResult asyncResult) [0x0002c] in <91935ad653254a93b9d73a9f8f2f7a2d>:0 
  at System.Net.Sockets.SocketTaskExtensions+<>c.<ConnectAsync>b__2_1 (System.IAsyncResult asyncResult) [0x00006] in <91935ad653254a93b9d73a9f8f2f7a2d>:0 
  at System.Threading.Tasks.TaskFactory`1[TResult].FromAsyncCoreLogic (System.IAsyncResult iar, System.Func`2[T,TResult] endFunction, System.Action`1[T] endAction, System.Threading.Tasks.Task`1[TResult] promise, System.Boolean requiresSynchronization) [0x00019] in <254335e8c4aa42e3923a8ba0d5ce8650>:0 
--- End of stack trace from previous location where exception was thrown ---

**Comment by o0donder0o** [comment]
Ok, got it working with a workaround.

Access to Spotweb from sonarr/radarr would just not work when accessing spotweb, I believe this is related to sonarr/radarr as jgeusebroek already implied.

As a workaround i've installed nzhydra. I've configured spotweb as an indexer and set the rev proxy address. Still no go. Probably because i've applied the setting i referred to in the previouse comment. 

I've removed this setting and configurred nzbhydra to connect over http (acceptable risk) and direct address (ip-address). Not NZBhrydra would download nzb files and add them to my usenet client. 

Just add nzbhydra to sonarr/radarr as the indexer and i will work in sonarr/radarr as well. 

Downsides
1. you'll loose insights what indexer (i've you use more indexers then spotweb) is used in sonarr/radarr since you'll only see nzbhydra.
2. When accessing Spotweb through rev proxy the interface is messed up, so access it by ip directly and over http. Might be be solved by applying the fix in the article from rev proxy side. Since i use traefik i could't get this to work.
